src/tests/unit/platform/test_admin_grants.py

```diff
@@ -0,0 +1,209 @@
+        grant.save()
+
+
+def test_create_profile_from_template(cbcsdk_mock):
+    """Test the creation of a new profile within a grant via a template."""
+    def respond_to_profile_grant(url, body, **kwargs):
+        ret = copy.deepcopy(POST_PROFILE_IN_GRANT_RESP)
+        ret['profile_uuid'] = body['profile_uuid']
+        return ret
+
+    cbcsdk_mock.mock_request('GET', '/access/v2/orgs/test/grants/psc:user:12345678:ABCDEFGH', GET_GRANT_RESP)
+    cbcsdk_mock.mock_request('POST', '/access/v2/orgs/test/grants/psc:user:ABC12345:DEF67890/profiles',
+                             respond_to_profile_grant)
+    api = cbcsdk_mock.api
+    grant = Grant(api, 'psc:user:12345678:ABCDEFGH')
+    template = {
+        "profile_uuid": "to-be-deleted",
+        "orgs": {
+            "allow": [
+                "psc:org:test2"
+            ],
+        },
+        "roles": [
+            "psc:role::SECOPS_ROLE_MANAGER"
+        ],
+        "conditions": {
+            "expiration": 0,
+            "disabled": False
+        }
+    }
+    profile = grant.create_profile(template)
+    profile.save()
+    assert profile.orgs['allow'] == ["psc:org:test2"]
+
+
+def test_modify_profile_within_grant(cbcsdk_mock):
+    """Test the modification of a profile within a grant."""
+    cbcsdk_mock.mock_request('GET', '/access/v2/orgs/test/grants/psc:user:12345678:ABCDEFGH', GET_GRANT_RESP)
+    cbcsdk_mock.mock_request('PUT', '/access/v2/orgs/test/grants/psc:user:12345678:ABCDEFGH/profiles/c57ba255-1736-4bfa-a59d-c54bb97a41d6',  # noqa: E501
+                             PUT_PROFILE_RESP)
+    api = cbcsdk_mock.api
+    grant = Grant(api, 'psc:user:12345678:ABCDEFGH')
+    profile = grant.profiles_[0]
+    profile.orgs['allow'].append('psc:org:test22')
+    profile.touch()
+    profile.save()
+    assert profile.orgs['allow'] == ["psc:org:test2", 'psc:org:test22']
+
+
+def test_delete_profile_within_grant(cbcsdk_mock):
+    """Test the deletion of a profile within a grant."""
+    cbcsdk_mock.mock_request('GET', '/access/v2/orgs/test/grants/psc:user:12345678:ABCDEFGH', GET_GRANT_RESP)
+    cbcsdk_mock.mock_request('DELETE', '/access/v2/orgs/test/grants/psc:user:12345678:ABCDEFGH/profiles/c57ba255-1736-4bfa-a59d-c54bb97a41d6',  # noqa: E501
+                             DELETE_PROFILE_RESP)
+    api = cbcsdk_mock.api
+    grant = Grant(api, 'psc:user:12345678:ABCDEFGH')
+    profile = grant.profiles_[0]
+    profile.delete()
+    assert profile.conditions['disabled'] is True
+
+
+def test_delete_grant(cbcsdk_mock):
+    """Test the deletion of a grant."""
+    cbcsdk_mock.mock_request('GET', '/access/v2/orgs/test/grants/psc:user:12345678:ABCDEFGH', GET_GRANT_RESP)
+    cbcsdk_mock.mock_request('DELETE', '/access/v2/orgs/test/grants/psc:user:12345678:ABCDEFGH', DELETE_GRANT_RESP)
+    api = cbcsdk_mock.api
+    grant = Grant(api, 'psc:user:12345678:ABCDEFGH')
+    grant.delete()
+    assert grant.revoked is True
+
+
+def test_query_grants(cbcsdk_mock):
```

Comment:
I don't see a test for getting a Grant by ID using the select like so `api.select(Grant, 'principal_urn')`

Reply:
That's because there isn't one. Does there need to be one, given that the code it would run has already been adequately covered by other tests?

Reply:
You could adjust `test_get_and_set_grant` to use this method to get this usecase

Reply:
OK